spec/slim_lint/linter/control_statement_spacing_spec.rb

```diff
@@ -0,0 +1,177 @@
+
+    it { should report_lint }
+  end
+
+  context 'element too much space after =' do
+    let(:slim) { 'div=  bad' }
+
+    it { should report_lint }
+  end
+
+  context 'id missing space before =' do
+    let(:slim) { '#submit= bad' }
+
+    it { should report_lint }
+  end
+
+  context 'id missing space after =' do
+    let(:slim) { '#submit =bad' }
+
+    it { should report_lint }
+  end
+
+  context 'id missing space around =' do
+    let(:slim) { '#submit=bad' }
+
+    it { should report_lint }
+  end
+
+  context 'id and class missing space around =' do
+    let(:slim) { '.some-class#submit=bad' }
+
+    it { should report_lint }
+  end
+
+  context 'id too much space before =' do
+    let(:slim) { '#submit  =bad' }
+
+    it { should report_lint }
+  end
+
+  context 'id too much space after =' do
+    let(:slim) { '#submit=  bad' }
+
+    it { should report_lint }
+  end
+
+  context 'class missing space before =' do
+    let(:slim) { '.klass= bad' }
+
+    it { should report_lint }
+  end
+
+  context 'class missing space after =' do
+    let(:slim) { '.klass =bad' }
+
+    it { should report_lint }
+  end
+
+  context 'class missing space around =' do
+    let(:slim) { '.klass=bad' }
+
+    it { should report_lint }
+  end
+
+  context 'class too much space before =' do
+    let(:slim) { '.klass  =bad' }
+
+    it { should report_lint }
+  end
+
+  context 'class too much space after =' do
+    let(:slim) { '.klass=  bad' }
+
+    it { should report_lint }
+  end
+
+  context 'class with hyphen missing space before =' do
+    let(:slim) { '.some-klass= bad' }
+
+    it { should report_lint }
+  end
+
+  context 'class with hyphen missing space after =' do
+    let(:slim) { '.some-klass =bad' }
+
+    it { should report_lint }
+  end
+
+  context 'class with hyphen missing space around =' do
+    let(:slim) { '.some-klass=bad' }
+
+    it { should report_lint }
+  end
+
+  context 'class with hyphen too much space before =' do
+    let(:slim) { '.some-klass  =bad' }
+
+    it { should report_lint }
+  end
+
+  context 'class with hyphen too much space after =' do
+    let(:slim) { '.some-klass=  bad' }
+
+    it { should report_lint }
+  end
+
+  context 'ruby code that contains a properly formatted equal sign' do
+    let(:slim) { 'div =bad = 1' }
+
+    it { should report_lint }
+  end
+
+  context 'ruby code that contains a properly formatted equal sign' do
+    let(:slim) { 'div= bad = 1' }
+
+    it { should report_lint }
+  end
+
+  context 'ruby code that contains a properly formatted equal sign' do
+    let(:slim) { 'div  = bad = 1' }
+
+    it { should report_lint }
+  end
+
+  # OK
+
+  context 'ruby code that contains an equal sign without spacing' do
+    let(:slim) { 'div = ok=1' }
+
+    it { should_not report_lint }
+  end
+
+  context 'element with hyphen' do
+    let(:slim) { 'div - ok' }
+
+    it { should_not report_lint }
+  end
+
+  context 'control statement without element' do
+    let(:slim) { '= ok' }
```

Comment:
I notice this doesn't have any opinion on `=ok` which, according to my source files, does not seem to raise an error. In my opinion, `= ok` (with the space) should be enforced. Any opinions from the author or @sds?

Reply:
I consequently use the space as it's more readable. Slim's [official snippets](http://slim-lang.com/) also keep that standard. I thought this was already enforced 😄 

I just checked, you're right and it is not enforced, the following test passes but should not:

```ruby
  context 'control statement without element and without space' do
    let(:slim) { '=ok' }

    it { should_not report_lint }
  end
```